### PR TITLE
Update to arrow `33.0.0`

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #.

# Rationale for this change

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/core/tests/sql/set_variable.rs [hunk 1]
```diff
@@ -412,14 +413,8 @@ async fn set_time_zone_bad_time_zone_format() {
...
-    let expected = vec![
-        "+-----------------------------------------------------+",
-        "| Utf8(\"2000-01-01T00:00:00\")                         |",
-        "+-----------------------------------------------------+",
-        "| 2000-01-01T00:00:00 (Unknown Time Zone '+08:00:00') |",
-        "+-----------------------------------------------------+",
-    ];
-    assert_batches_eq!(expected, &result);
+    let err = pretty_format_batches(&result).err().unwrap().to_string();
+    assert_eq!(err, "Parser error: Invalid timezone \"+08:00:00\": Expected format [+-]XX:XX, [+-]XX, or [+-]XXXX");
```
[the remark this answers is not in the document]
Comment (alamb) [reply datafusion/core/tests/sql/set_variable.rs:417]: I think it is ok -- cc @comphead  and @waitingkuo

**Review by alamb — COMMENTED** [review]
Thanks @tustvold

**Comment by tustvold** [comment]
```
   process didn't exit successfully: `/Users/runner/work/arrow-datafusion/arrow-datafusion/target/debug/deps/datafusion-ba45d330e390f3a9` (signal: 4, SIGILL: illegal instruction)
```

Eep, will investigate, shenanigans may be afoot

**Comment by tustvold** [comment]
Shenanigans did abound - https://github.com/apache/arrow-rs/issues/3691, fix in https://github.com/apache/arrow-rs/pull/3692

**Comment by alamb** [comment]
> Shenanigans did abound - https://github.com/apache/arrow-rs/issues/3691, fix in https://github.com/apache/arrow-rs/pull/3692

Testing for the win!

File: datafusion/core/src/physical_plan/aggregates/mod.rs [hunk 2]
```diff
@@ -811,22 +811,22 @@ mod tests {
...
-            "| 3 | 2 | 2               |",
-            "| 3 | 3 | 1               |",
-            "| 4 |   | 3               |",
-            "| 4 | 3 | 1               |",
-            "| 4 | 4 | 2               |",
-            "+---+---+-----------------+",
+            "+---+-----+-----------------+",
+            "| a | b   | COUNT(1)[count] |",
+            "+---+-----+-----------------+",
+            "|   | 1.0 | 2               |",
```
Comment (alamb) [inline datafusion/core/src/physical_plan/aggregates/mod.rs:817]: these changes are going to create some serious downstream churn I suspect. I wonder if we should wait until https://github.com/apache/arrow-rs/pull/3717 is released (so downstream crates can choose to have the old behavior) 🤔

**Review by alamb — APPROVED** [review]
I am ok with these changes, but I worry about the possible effects on our downstream users (e.g. @maxburke  was just discussing the upgrade pain earlier today on ASF slack).

What would you think about sending a note to the mailing lists / slack channels giving a heads up and asking for feedback? I can do the communicating if you like

File: datafusion/core/src/physical_plan/aggregates/mod.rs (hunk 2 above)
Comment (comphead) [reply datafusion/core/src/physical_plan/aggregates/mod.rs:817]: @alamb is the issue related to  column name `[count]`?

**Comment by tustvold** [comment]
> I can do the communicating if you like

Yes please, my 2 cents is optimising for stability is premature at this point, but welcome other input

File: datafusion/core/tests/sql/functions.rs [hunk 3]
```diff
@@ -67,8 +67,8 @@ async fn csv_query_cast_literal() -> Result<()> {
         "+--------------------+----------+",
         "| c12                | Int64(1) |",
         "+--------------------+----------+",
-        "| 0.9294097332465232 | 1        |",
-        "| 0.3114712539863804 | 1        |",
+        "| 0.9294097332465232 | 1.0      |",
```
Comment (liukun4515) [inline datafusion/core/tests/sql/functions.rs:70]: I am confused about this changes？
cc @alamb @tustvold 
As you said in https://github.com/apache/arrow-datafusion/pull/5241/files#r1106342898.

Comment (tustvold) [reply datafusion/core/tests/sql/functions.rs:70]: What are you confused about? Despite its name, this column is a floating point type and so is formatted with a decimal point?

File: datafusion/core/src/physical_plan/aggregates/mod.rs (hunk 2 above)
Comment (tustvold) [reply datafusion/core/src/physical_plan/aggregates/mod.rs:817]: Coming back to this, even with https://github.com/apache/arrow-rs/pull/3717 floats will be formatted with a decimal point, there isn't an option currently exposed to configure this...

File: datafusion/core/tests/sql/functions.rs (hunk 3 above)
Comment (alamb) [reply datafusion/core/tests/sql/functions.rs:70]: > Despite its name

I certainly found that confusing 😆

**Comment by alamb** [comment]
I posted a note to the mailing list: https://lists.apache.org/thread/9y6bhdj2vgo9ll7bj72mf5gpwxkqy2b1

File: datafusion/core/tests/sql/functions.rs (hunk 3 above)
Comment (liukun4515) [reply datafusion/core/tests/sql/functions.rs:70]: > What are you confused about? Despite its name, this column is a floating point type and so is formatted with a decimal point?

I missed the data type from the name of this column

**Comment by alamb** [comment]
Unless there are any comments, I think we should merge this next week

File: datafusion/core/tests/sql/set_variable.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (waitingkuo) [reply datafusion/core/tests/sql/set_variable.rs:417]: sorry for the late reply. this looks great to me. I originally followed how pyarrow works to implement it. raising parser error makes more sense to me.

**Pull request merged**